### PR TITLE
save form teacher input on database

### PR DESCRIPTION
**Pull request by malkitbenning**
Not yet ready for review - still developing

### CODE REVIEW

**Comment by malkitbenning** [comment]
ready for review

File: client/src/components/data/UserFormInput.json
Comment (m4ndycheung) [inline client/src/components/data/UserFormInput.json:1]: Review: Matches the JSON data structure discussed in dev team meeting, no issues here

File: server/save-user-form-input.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        res.status(502).json({
+          result: "failure",
+          message: "Error saving teacher answers",
+        });
+        client.end();
+      });
+  }
+
+  const { formSubmission } = req.body;
+  const { teacherID, pupilID, pupilName, updateDate, overrideScore, overrideComment, teacherAnswers } = formSubmission;
```
Comment (m4ndycheung) [inline server/save-user-form-input.js:33]: Question: In this object destructuring, I'm curious if teacherAnswers should be questionAnswers to match the key in UserFormInput.json? Or if questionAnswers in the json should be changed to teacherAnswers?

JSON data
{
"formSubmission": {
"pupilID": 7,
"teacherID": 1,
"pupilName": "Jimmy",
"updateDate": "23.10.2023",
"overrideScore": 73,
"overrideComment": "Experience",
"questionsAnswers": [
{
"questionID": 1,
"answerID": 4,
"comment": "This is my comment"
},
{
"questionID": 2,
"answerID": 8
}
]
}
}

**Review by m4ndycheung — CHANGES_REQUESTED** [review]
Hey @malkitbenning the code makes sense to me, you've done a stellar job figuring out how to save the user form input data. I have one question for you, otherwise, looks good to merge.

File: server/save-user-form-input.js (hunk 1 above)
Comment (malkitbenning) [reply server/save-user-form-input.js]: sorted after review - changed the name

**Pull request merged**